unittests/AST/ASTImporterTest.cpp

```diff
@@ -4012,6 +4024,36 @@ TEST_P(ASTImporterLookupTableTest,
   EXPECT_EQ(Res.size(), 0u);
 }
 
+TEST_P(ASTImporterLookupTableTest,
+       LookupFindsFriendClassDeclWithTypeAlias) {
+  TranslationUnitDecl *ToTU = getToTuDecl(
+      R"(
+      class F;
+      using alias_of_f = F;
+      class Y { friend alias_of_f; };
+      )",
+      Lang_CXX11);
+
+  ASTImporterLookupTable LT(*ToTU);
+
+  auto *FriendD = FirstDeclMatcher<FriendDecl>().match(ToTU, friendDecl());
+  auto *Y = FirstDeclMatcher<CXXRecordDecl>().match(ToTU, cxxRecordDecl(hasName("Y")));
+  auto *F = FirstDeclMatcher<CXXRecordDecl>().match(ToTU, cxxRecordDecl(hasName("F")));
+
+  const RecordDecl *RD = getRecordDeclOfFriend(FriendD);
+
+  DeclarationName FwdName = F->getDeclName();
+  auto Res = LT.lookup(ToTU, FwdName);
+  EXPECT_EQ(Res.size(), 1u);
+  EXPECT_EQ(*Res.begin(), RD);
+
+  Res = LT.lookup(Y, FwdName);
+  EXPECT_EQ(Res.size(), 0u);
+
+  DeclarationName AliasName = RD->getDeclName();
```

Comment:
Checking the names is not so important here, I think.
Probably, instead, we can just add after line 4043:
`EXPECT_EQ(F, RD)`